tests/components/freedompro/test_switch.py

```diff
@@ -0,0 +1,83 @@
+"""Tests for the Freedompro switch."""
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN, SERVICE_TURN_ON
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_TURN_OFF, STATE_OFF
+from homeassistant.helpers import entity_registry as er
+
+
+async def test_switch_get_state(hass, init_integration):
```

Comment:
Update the test to verify the state changes when you got an updated state from the API